### ISSUE TITLE
phantomsetup gives weird results for default shock example problem

### ISSUE THREAD

**Issue by danieljprice** [danieljprice/phantom#47]
something in the initial conditions is screwy here, the initial density profile is not correct when setup with default answers to phantomsetup (which should run the Sod problem).

Looks like an issue with the "smoothed initial conditions" stretch mapping being activated by default, even though no smoothing should occur for this problem.

![splash](https://user-images.githubusercontent.com/12252103/89760772-63f2ef80-db30-11ea-8969-2072442cf431.png)

**Comment by jameswurster** [comment]
I just took a look at this, and found a few issues, not all of which agree with you. 
1) Using SETUP=shock and all the defaults, the .tmp file yields the discontinuous density profile that I expect for me
2) It looks like a smoothing function over the discontinuity is begin called (i.e. rhosmooth is always being called from set_shock)
3) The values in the .in file are wrong.  We define many of these values in choose_shock which is only run via the interactive method; since phantomsetup stops before the .in file is written in this case, then these values are not set when starting from the .setup file, yield the wrong parameters (this issue will also affect setup_sphereinbox as well, since the size of the sink particle is set interactively)

**Comment by danieljprice** [comment]
yep:
1. the weird density is only once you run through phantom to compute density the first time
2. this is a feature I implemented, but should result in no smoothing if the parameter is zero, instead it looks like it does something weird. So likely a bug I've introduced.
3. agreed, I also noticed this. To solve this I think best would be to add the choice of tmax and dtmax to the .setup file, so the interactive/non-interactive setups do the same thing. Or another possibility would be to take out the option of choosing pre-selected shock tube problems altogether and instead distribute .in and .setup files for particular problems in an examples repo.

**Comment by jameswurster** [comment]
I have been able to reproduce the error.  In the function fsmooth, the option for fac <= 0 should read
`    delta = 2.*sign(1.0,x-x0)*dsmooth`
(i.e. the arguments of the sign function are in the wrong order)   I have not committed this fix since you would need to approve the pull request anyway.

Regarding point 3:
I considered this when I was adding the prompt to the sink radius in sphereinbox, but opted not to since I thought it might get confusing with the same variables defined in both .in and .setup.  Plus for me it was a simple patch to not stop phantomsetup after the prompts.  Despite this, it might be simplest to add another block to the setup file where we write any of the .in parameters.  While I like the idea of pre-stocked .in & .setup for simple & test cases, I still like the idea of being able to generate them all via the prompts without having to select and/or edit the relevant text files.

**Comment by jameswurster** [comment]
I just committed the bug patch for this issue.
It can now be closed.  I will open a new issue regarding the discussion of the .in and .setup files.
